Assembler.UnitTests/Releasing/AssemblingPositionCountValidatorsTests.cs

```diff
@@ -0,0 +1,69 @@
+using Assembler.Core.Enums;
+using Assembler.Core.RawAssemblingEntities;
+using Moq;
+using NUnit.Framework;
+
+namespace Assembler.UnitTests.Releasing
+{
+    [TestFixture]
+    public class AssemblingPositionCountValidatorsTests
+    {
+        [TestCaseSource(nameof(GetObjects))]
+        public void IsValid_VariousInputs_VariousExpectedOutputs(IValidator<RawMessageInAssembly> validator,
+             int numberOfInitialFrames, int numberOfMiddleFrames, int numberOfFinalFrames, bool expectedResult)
+        {
+            // Arrange
+            var message = new RawMessageInAssembly(DateTime.MinValue, DateTime.MinValue);
+
+            for (int i = 0; i < numberOfInitialFrames; i++)
+            {
+                message.InitialFrames.Add(new Mock<BaseFrame>(AssemblingPosition.Initial).Object);
+            }
+
+            for (int i = 0; i < numberOfMiddleFrames; i++)
+            {
+                message.MiddleFrames.Add(new Mock<BaseFrame>(AssemblingPosition.Middle).Object);
+            }
+
+            for (int i = 0; i < numberOfFinalFrames; i++)
+            {
+                message.FinalFrames.Add(new Mock<BaseFrame>(AssemblingPosition.Final).Object);
+            }
+
+            // Act
+            var result = validator.IsValid(message);
+
+            // Assert
+            Assert.AreEqual(expectedResult, result);
+        }
+
+        private static object[] GetObjects()
+        {
+            var minimumFrames = 2;
+
+            var initialValidator = new InitialAssemblingPositionCountValidator(minimumFrames, Utilities.GetLoggerFactory());
+            var middleValidator = new MiddleAssemblingPositionCountValidator(minimumFrames, Utilities.GetLoggerFactory());
+            var finalValidator = new FinalAssemblingPositionCountValidator(minimumFrames, Utilities.GetLoggerFactory());
+
+            return new object[]
+            {
+                new TestCaseData(initialValidator, 0, 0, 0, false).SetName("InitialWith0"),
+                new TestCaseData(initialValidator, 1, 0, 0, false).SetName("InitialWith1"),
+                new TestCaseData(initialValidator, 2, 0, 0, true).SetName("InitialWith2"),
+                new TestCaseData(initialValidator, 100, 0, 0, true).SetName("InitialWith100"),
+                new TestCaseData(middleValidator, 0, 0, 0, false).SetName("MiddleWith0"),
+                new TestCaseData(middleValidator, 0, 1, 0, false).SetName("MiddleWith1"),
+                new TestCaseData(middleValidator, 0, 2, 0, true).SetName("MiddleWith2"),
+                new TestCaseData(middleValidator, 0, 100, 0, true).SetName("MiddleWith100"),
+                new TestCaseData(finalValidator, 0, 0, 0, false).SetName("FinalWith0"),
+                new TestCaseData(finalValidator, 0, 0, 1, false).SetName("FinalWith1"),
+                new TestCaseData(finalValidator, 0, 0, 2, true).SetName("FinalWith2"),
+                new TestCaseData(finalValidator, 0, 0, 100, true).SetName("FinalWith100"),
```

Comment:
I dont know why (I installed .net core 2.2 and 3.1), the .SetName making this test to be inconclusive and therefore NUnit doesnt run it.... When I removed it it worked and the tests passed.

Are you not having the same problem?

Reply:
No, I checked with their site because I had some problems with object[] and then changed it according to their docs